diagnostics/forcing_feedback/obs_processing_script.py

```diff
@@ -0,0 +1,173 @@
+# feedback_regress
+#
+# Regeresses radiative flux anomalies with global-mean dTs to compute 2D feedback
+def feedback_regress(fluxanom,tspert,tsclimo,lat,lon,fbname):
+
+    sp = tspert.shape
+    sc = tsclimo.shape
+
+    tsclimo_re = np.squeeze(np.nanmean(np.reshape(tsclimo, \
+                (np.int(sc[0]/12),12,sc[1],sc[2])),axis=0))
+
+    tsanom = tspert - np.reshape(np.repeat(tsclimo_re[np.newaxis,...],np.int(sp[0]/12), \
+               axis=0),(sp[0],sp[1],sp[2]))
+
+    weights = np.repeat(np.cos(np.deg2rad(lat))[np.newaxis,:],sp[0],axis=0)
+    tsanom_globemean = globemean_3D(tsanom,weights)
+    tsanom_re = np.repeat(np.repeat(tsanom_globemean[:,np.newaxis], \
+                          sp[1],axis=1)[...,np.newaxis],sp[2],axis=2)
+
+    tsanom_re_timemean = np.nanmean(tsanom_re,axis=0)
+    tsanom_re_std = np.nanstd(tsanom_re,axis=0)
+    fluxanom_timemean = np.nanmean(fluxanom,axis=0)
+    fluxanom_std = np.nanstd(fluxanom,axis=0)
+
+    n=np.sum(~np.isnan(tsanom_re),axis=0)
+    cov = np.nansum((tsanom_re-tsanom_re_timemean)*\
+             (fluxanom-fluxanom_timemean),axis=0)/n
+    slopes = cov/(tsanom_re_std**2)
+
+    slopes = xr.DataArray(slopes,coords=[lat,lon],dims=['lat','lon'],name=fbname)
+
+    return slopes
+
+
+kname = 'CloudSat'
+regime = 'TOA'
+
+#Read in reanalysis data and kernel-derived radiative anomalies and forcing, which were computed using a slightly modified version of the MDTF "Forcing and Feedback" module code. Then, regress against global-mean surface temperature, when applicable, to compute radiative feedbacks using the "feedback_regress" function detailed above. For Instantaneous Radiative Forcing, we just regress against time (trend)
+
+#Surface temperature
+ts = xr.open_dataset('/gpfsm/dnb32/rjkramer/ERAfb/temp/file_out_ts.nc')
+ts = ts.assign_coords(time=pd.date_range('2003-01',periods=len(ts.time.values),freq='M'))
+ts = ts.sel(time=slice('2003-01','2018-12')) #Maximum possible timeseries given other data
+ts_pert = ts.t2.values
+ts_climo = ts.t2.values
+ts = None
+
+#Planck radiative anomalies
+inputfile = '/gpfsm/dnb32/rjkramer/ERAfb/results/'+regime+'_Fluxanom_planck_Results_K'+kname+'_ERAfb_allERA_retry.nc'
+nc_pl_era = xr.open_dataset(inputfile)#'/gpfsm/dnb32/rjkramer/ERAfb/results/file_out1.nc')
+nc_pl_era = nc_pl_era.assign_coords(time=pd.date_range('2003-01',periods=len(nc_pl_era.time.values),freq='M'))
+#nc_pl_era = nc_pl_era.sel(latitude=slice(-60,60))
+inputfile = None
+
+varhold = nc_pl_era.fluxanom_pl_sfc_tot.values+nc_pl_era.fluxanom_pl_trop_tot.values+nc_pl_era.fluxanom_pl_strat_tot.values
+fb_pl_era = feedback_regress(varhold,ts_pert,ts_climo,nc_pl_era.latitude.values,nc_pl_era.longitude.values,'Planck')
+varhold = None
+
+#Lapse Rate radiative anomalies
+inputfile = '/gpfsm/dnb32/rjkramer/ERAfb/results/'+regime+'_Fluxanom_lapserate_Results_K'+kname+'_ERAfb_allERA_retry.nc'
+nc_lr_era = xr.open_dataset(inputfile)#'/gpfsm/dnb32/rjkramer/ERAfb/results/file_out2.nc')
```

Comment:
remove comment with hard-coded path

Reply:
Thanks @wrongkindofdoctor for catching all of these! The obs_processing_script.py, where most of your hard coding-related comments come from, is just the script I used to generate the digested observational data. Since it's independent of the actual module, is there a certain formatting I should use instead of including my actual, hard coded paths?  Replace them with dummy paths of some sort? Thanks for the clarification.

Reply:
@briansoden  Ah, I see. I have no issue with keeping the comments for the paths as-is, then.